### PR TITLE
Add the option to add text before and after help output

### PR DESCRIPTION
**Pull request by CrustyAuklet**
It would be nice to add text to the beginning and end of help messages. Many standard *nix tools do this to provide examples, show license information, etc.

For example, the current help output on a firmware programmer I am building:
```
Usage: dev-program [options] FILE

Positional arguments:
FILE                    HEX input file

Optional arguments:
-h --help               show this help message and exit
-l --list-interfaces    List all possible programming interfaces detectable by this program
-d --delay              Amount of time to wait for a device reset for programming, in milliseconds (default 5000)
-r --auto-reset         Attempt to automatically reset the device (depends on programming cable and/or current firmware)
```

Turns into:
```
Usage: dev-program [options] FILE
Internal Device Programmer Version 0.1.0

Examples:
    dev-program -w -v image.hex  # Write image.hex to the device and verify
    dev-program -l               # List all potential programming devices

Positional arguments:
FILE                    HEX input file

Optional arguments:
-h --help               show this help message and exit
-l --list-interfaces    List all possible programming interfaces detectable by this program
-d --delay              Amount of time to wait for a device reset for programming, in milliseconds (default 5000)
-r --auto-reset         Attempt to automatically reset the device (depends on programming cable and/or current firmware)

Please contact HelpDesk@company.com with any questions
Submit any bugs at https://gitlab.company.com/dev-program/issues
```

To add this I added two new string members, and printed them out if they are not empty during the help output. I tried to make sure the formatting remains unchanged if the text is not used. The spacing when used I attempted to make it look like the output of `ls -help` and others.

I did this quickly because I needed it for this current project, so totally open to any suggestions to make it better, or rejection if you don't think it is in scope for your project.

### CODE REVIEW

File: include/argparse.hpp [hunk 1]
```diff
@@ -800,6 +800,14 @@ class ArgumentParser {
...
   }
 
+  void add_pre_text(std::string aPreText) {
+    mPreText = std::move(aPreText);
+  }
+
+  void add_post_text(std::string aPostText) {
+    mPostText = std::move(aPostText);
+  }
+
```
Comment (p-ranav) [inline include/argparse.hpp]: Similar to Python, let's change these method names (and members) to [description](https://docs.python.org/2/library/argparse.html#description) and [epilog](https://docs.python.org/2/library/argparse.html#)

`parser.add_description(value)` and `parser.add_epilog(value)` 

* `description`:  This argument gives a brief description of what the program does and how it works. In help messages, the description is displayed between the command-line usage string and the help messages for the various arguments.
* `epilog`: Some programs like to display additional description of the program after the description of the arguments. Such text can be specified using epilog.

**Comment by CrustyAuklet** [comment]
In my haste I didn't think of the obvious, and check against python argparse. looking now I noticed that python argparse prints the description like this:
```
usage: argparse.py [-h]

A foo that bars

optional arguments:
 -h, --help  show this help message and exit

And that's how you'd foo a bar
```

currently I am missing the space above the description because I modeled it after `ls`:
```
Usage: ls [OPTION]... [FILE]...
List information about the FILEs (the current directory by default).
Sort entries alphabetically if none of -cftuvSUX nor --sort is specified.

Mandatory arguments to long options are mandatory for short options too.
  -a, --all                  do not ignore entries starting with .
  -A, --almost-all           do not list implied . and ..
```
Would you like the extra space to make it more like the python output?

**Comment by p-ranav** [comment]
Yeah let's add it. Thanks!

**Comment by p-ranav** [comment]
Thanks for your contribution!

**Pull request merged**